src/pypdfium2/_helpers/page.py

```
@@ -336,6 +340,131 @@ def flatten(self, flag=pdfium_c.FLAT_NORMALDISPLAY):
+                The dictionary will be empty if all fonts are embedded or if no text objects are found.
+        
+        Example:
+            >>> missing = page.get_missing_fonts()
+            >>> if missing:
+            ...     print(f"Found {len(missing)} non-embedded fonts")
+            
+            >>> # Reuse an existing textpage
+            >>> textpage = page.get_textpage()
+            >>> missing = page.get_missing_fonts(textpage=textpage)
+            >>> textpage.close()
+        """
+        missing_fonts = {}
+        textpage_created = False
+        try:
+            if textpage is None:
+                textpage = self.get_textpage()
+                textpage_created = True
+            
+            for obj in self.get_objects(filter=[pdfium_c.FPDF_PAGEOBJ_TEXT], textpage=textpage):
+                try:
+                    font = obj.get_font()
+                    if not font.get_is_embedded():
+                        font_name = font.get_base_name()
+                        
+                        text_content = None
+                        try:
+                            extracted = obj.extract()
+                            if extracted and extracted.strip():
+                                if len(extracted) > 100:
+                                    text_content = extracted[:97] + "..."
+                                else:
+                                    text_content = extracted
+                        except (PdfiumError, Exception):
+                            pass
+                        
+                        if font_name not in missing_fonts:
+                            missing_fonts[font_name] = []
+                        if text_content:
+                            missing_fonts[font_name].append(text_content)
+                except (PdfiumError, AttributeError):
+                    continue
+        except Exception:
+            pass
```

I might prefer to avoid broad exception catching here.
In particular, I'd say `Exception` and `AttributeError` are too generic and could theoretically mask e.g. a typo in the code.
Also, I think there is some redundancy in your current try/except hierarchy.

---

Even with `PdfiumError`, I'm not sure silently suppressing it is good.
Unless there's a concrete reason (like with `textpage.get_textobj()` and escape chars), it might fit the API design better not to do any exception catching here, and leave it up to the caller whether they want to guard for `PdfiumError`.